### PR TITLE
Initialize App Check debug mode in initializeAppCheck

### PR DESCRIPTION
**Pull request by hsubox76**
Initialize debug mode for App Check (i.e., reading the global var) in the first call to `initializeAppCheck()` instead of when the app-check component is registered. This makes it easier for developers to set the variable in code.

Additionally, it logs a `console.warn` on subsequent calls to initializeAppCheck notifying the user they are still in debug mode, and logging the token (if they don't reload/quit before that token promise resolves). This may help HMR users who lose the initial token message.

### CODE REVIEW

File: packages/app-check/src/api.ts [hunk 1]
```diff
@@ -73,6 +90,9 @@ export function initializeAppCheck(
     }
   }
 
+  // Only read global variable on first call to `initializeAppCheck()`.
+  initializeDebugMode();
```
Comment (Feiyang1) [inline packages/app-check/src/api.ts:94]: It should be called only once for all app instances, not per app instance.

File: packages/app-check/src/api.ts [hunk 2]
```diff
@@ -65,6 +67,21 @@ export function initializeAppCheck(
...
+        logger.warn(
+          `App Check is in debug mode. To turn off debug mode, unset ` +
+            `the global variable FIREBASE_APPCHECK_DEBUG_TOKEN and ` +
+            `restart the app.`
+        );
+        // Make this a separate console statement so user will at least have the
+        // first message if the token promise doesn't resolve in time.
+        void getDebugToken().then(token =>
+          logger.warn(`Debug token is ${token}.`)
+        );
```
Comment (Feiyang1) [inline packages/app-check/src/api.ts:83]: We should print the debug token every time when `initializeAppCheck` is called if it's in the debug mode, not just the second time. For example, print the token after the page is refreshed.

**Review by Feiyang1 — CHANGES_REQUESTED** [review]

File: packages/app-check/src/api.ts (hunk 2 above)
Comment (hsubox76) [reply packages/app-check/src/api.ts:83]: Ok, can do, just FYI, if it's the first call to initializeApp(), the user will see
`App Check debug token: ${newToken}. You will need to add it to your app's App Check settings in the Firebase console for it to work`
And then they will also see these 2 messages right after that. Is that a bit clunky?

I guess the exception is if they have 2 FirebaseApps, then they'll only see all 3 messages when calling initializeApp for the first FirebaseApp.

Comment (Feiyang1) [reply packages/app-check/src/api.ts]: `logger.warn(`Debug token is ${token}.`)` is not necessary, if we already print `App Check debug token: ${newToken}. You will need to add it to your app's App Check settings in the Firebase console for it to work`.

Comment (Feiyang1) [reply packages/app-check/src/api.ts]: so no matter whether it's the first or nth time, always show the same message.

**Comment by rickstaa** [comment]
@hsubox76 Thanks a lot for your work on this feature. Just a small heads-up. I noticed that in the old version, the Debug token is not printed to the console in Firefox 92.0 (64-bit) on Linux. It is easily found in the `firebase-app-check-db` though. On chrome, everything works fine.

**Comment by rickstaa** [comment]
@hsubox76 I removed my previous comment since it was caused by some settings in Firefox.

File: packages/app-check/src/api.ts (hunk 2 above)
Comment (hsubox76) [reply packages/app-check/src/api.ts]: Ok, done, see if that looks acceptable now.

File: packages/app-check/src/api.ts (hunk 1 above)
Comment (hsubox76) [reply packages/app-check/src/api.ts]: Changed.

File: packages/app-check/src/api.ts [hunk 3]
```diff
@@ -67,14 +66,13 @@ export function initializeAppCheck(
   // Log a warning when `initializeAppCheck()` is called in debug mode,
```
Comment (Feiyang1) [inline packages/app-check/src/api.ts:66]: Comment needs update.

File: packages/app-check/src/api.test.ts [hunk 4]
```diff
@@ -118,6 +120,40 @@ describe('api', () => {
...
+        provider: new ReCaptchaV3Provider(FAKE_SITE_KEY)
+      });
+      // Ensure getDebugToken() call inside `initializeAppCheck()`
+      // has time to resolve, and double check its value matches that
+      // written to indexedDB.
+      expect(await getDebugToken()).to.equal(token);
+      expect(consoleStub.args[0][0]).to.include(token);
+      self.FIREBASE_APPCHECK_DEBUG_TOKEN = undefined;
+    });
+    it('warns about debug mode on second call', async () => {
```
Comment (Feiyang1) [inline packages/app-check/src/api.test.ts:143]: What does it test? 
I think we need a test to verify `initializeDebugMode` is called only once.

File: packages/app-check/src/api.ts (hunk 3 above)
Comment (hsubox76) [reply packages/app-check/src/api.ts:66]: Updated.

File: packages/app-check/src/api.test.ts (hunk 4 above)
Comment (hsubox76) [reply packages/app-check/src/api.test.ts]: Oh wow, the test caught the fact that I was not setting the `initialize` property... thanks.

**Review by Feiyang1 — APPROVED** [review]

**Comment by Feiyang1** [comment]
Please add a changeset. @hsubox76

**Pull request merged**